openupgrade_scripts/scripts/payment/15.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,61 @@
+from openupgradelib import openupgrade
+
+_renamed_fields = [
+    (
+        "payment.token",
+        "payment_token",
+        "payment_ids",
+        "transaction_ids",
+    ),
+    (
+        "payment.transaction",
+        "payment_transaction",
+        "is_processed",
+        "is_post_processed",
+    ),
+    (
+        "payment.transaction",
+        "payment_transaction",
+        "payment_token_id",
+        "token_id",
+    ),
+]
+
+_renamed_xmlids = [
+    ("payment.payment_token_salesman_rule", "payment.payment_token_billing_rule"),
+    ("payment.payment_acquirer_ingenico", "payment.payment_acquirer_ogone"),
+    ("payment.payment_acquirer_payu", "payment.payment_acquirer_payumoney"),
+]
+
+
+def fill_payment_token_name(env):
+    # field it's required now
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE payment_token
+        SET name = 'XXXXXXXXXXXX????'
+        WHERE name IS NULL""",
+    )
+
+
+def fill_payment_transaction_partner_id(env):
+    # field it's required now
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE payment_transaction pt
+        SET partner_id = ap.partner_id
+        FROM account_payment ap
+        WHERE ap.partner_id IS NOT NULL
```

Comment:
Can it be the partner_id in the payment null? If so, the required may still not be fulfilled. If not, this condition is useless.